crates/re_types_builder/src/objects.rs

```diff
@@ -0,0 +1,826 @@
+    /// ```flatbuffers
+    /// /// \py Something something about how this fields behave in python.
+    /// my_field: uint32,
+    /// ```
+    ///
+    /// See also [`Docs::doc`].
+    pub tagged_docs: HashMap<String, Vec<String>>,
+}
+
+impl Docs {
+    fn from_raw_docs(
+        docs: Option<flatbuffers::Vector<'_, flatbuffers::ForwardsUOffset<&'_ str>>>,
+    ) -> Self {
+        // language-agnostic docs
+        let doc = docs
+            .into_iter()
+            .flat_map(|doc| doc.into_iter())
+            // NOTE: discard tagged lines!
+            .filter(|line| !line.trim().starts_with('\\'))
+            .map(ToOwned::to_owned)
+            .collect::<Vec<_>>();
+
+        // tagged docs, e.g. `\py this only applies to python!`
+        let tagged_docs = {
+            let tagged_lines = docs
+                .into_iter()
+                .flat_map(|doc| doc.into_iter())
+                // NOTE: discard _un_tagged lines!
+                .filter_map(|line| {
+                    let trimmed = line.trim();
+                    trimmed.starts_with('\\').then(|| {
+                        let tag = trimmed.split_whitespace().next().unwrap();
+                        let line = &trimmed[tag.len()..];
+                        (tag[1..].to_owned(), line.to_owned())
+                    })
+                })
+                .collect::<Vec<_>>();
+
+            let all_tags: HashSet<_> = tagged_lines.iter().map(|(tag, _)| tag).collect();
+            let mut tagged_docs = HashMap::new();
+
+            for cur_tag in all_tags {
+                tagged_docs.insert(
+                    cur_tag.clone(),
+                    tagged_lines
+                        .iter()
+                        .filter_map(|(tag, line)| (cur_tag == tag).then(|| line.clone()))
+                        .collect(),
+                );
+            }
+
+            tagged_docs
+        };
+
+        Self { doc, tagged_docs }
+    }
+}
+
+/// A high-level representation of a flatbuffers object, which can be either a struct, a union or
+/// an enum.
+#[derive(Debug, Clone)]
+pub struct Object {
+    /// File path of the associated fbs definition, e.g. `//rerun/components/point2d.fbs`.
+    pub filepath: String,
+
+    /// Fully-qualified name of the object, e.g. `rerun.components.Point2D`.
+    pub fqname: String,
+
+    /// Fully-qualified package name of the object, e.g. `rerun.components`.
+    pub pkg_name: String,
+
+    /// Name of the object, e.g. `Point2D`.
+    pub name: String,
+
+    /// The object's multiple layers of documentation.
+    pub docs: Docs,
+
+    /// The object's kind: datatype, component or archetype.
+    pub kind: ObjectKind,
+
+    /// The object's attributes.
+    pub attrs: Attributes,
+
+    /// The object's inner fields, which can be either struct members or union values.
+    ///
+    /// These are pre-sorted, in ascending order, using their `order` attribute.
+    pub fields: Vec<ObjectField>,
```

Comment:
`fields_or_values`?

Reply:
The flatbuffers AST makes that distinction, but we don't: it's all an `ObjectField` as far as we're concerned once we're done with the semantic pass.